src/main/java/org/eclipsefoundation/marketplace/model/CacheableResource.java

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.model;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Interface denoting that a resource can be cached, and promises methods that
+ * aid in caching. This is meant to be used on REST resources to cache results
+ * of calls.
+ * 
+ * @author Martin Lowe
+ */
+public abstract class CacheableResource {
```

Comment:
What about removing this abstract class, and make the computation of the cache key an implementation of the GuavaCachingService? Or even better, make the cache key computation an injectable strategy of the GuavaCachingService.

I'm not a huge fan of introducing an abstract class to make a resource cacheable. Generally, try to favor composition over inheritance to introduce behavior.